internal/engine/exit/conv.go

```diff
@@ -0,0 +1,254 @@
+	tiltruns "github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
+	"github.com/tilt-dev/tilt/pkg/model"
+)
+
+func targetsForResource(mt *store.ManifestTarget, holds buildcontrol.HoldSet) []tiltruns.Target {
+	var resources []tiltruns.Target
+
+	buildResource := buildTarget(mt, holds)
+	if buildResource != nil {
+		resources = append(resources, *buildResource)
+	}
+
+	runtimeResource := runtimeTarget(mt, holds)
+	if runtimeResource != nil {
+		resources = append(resources, *runtimeResource)
+	}
+
+	return resources
+}
+
+func k8sRuntimeTarget(mt *store.ManifestTarget) *tiltruns.Target {
+	krs := mt.State.K8sRuntimeState()
+	if mt.Manifest.PodReadinessMode() == model.PodReadinessIgnore && krs.HasEverDeployedSuccessfully && len(krs.Pods) == 0 {
+		// HACK: engine assumes anything with an image will create a pod; PodReadinessIgnore is used in these
+		// 	instances to avoid getting stuck in pending forever; in reality, there's no "runtime" target being
+		// 	monitored by Tilt, so instead of faking it, just omit it (note: only applies AFTER first deploy so
+		// 	that we can determine there are no pods, so it will appear in waiting until then, which is actually
+		// 	desirable and matches behavior in K8sRuntimeState::RuntimeStatus())
+		// 	see https://github.com/tilt-dev/tilt/issues/3619
+		return nil
+	}
+
+	target := &tiltruns.Target{
+		Name:      fmt.Sprintf("%s:runtime", mt.Manifest.Name.String()),
+		Resources: []string{mt.Manifest.Name.String()},
+	}
+
+	if mt.Manifest.IsK8s() && mt.Manifest.K8sTarget().HasJob() {
+		target.Type = tiltruns.TargetTypeJob
+	} else {
+		target.Type = tiltruns.TargetTypeServer
+	}
+
+	// a lot of this logic is duplicated from K8sRuntimeState::RuntimeStatus()
+	// but ensures Job containers are handled correctly and adds additional
+	// metadata
+	pod := krs.MostRecentPod()
+	switch pod.Phase {
+	case v1.PodRunning:
+		target.State.Active = &tiltruns.TargetStateActive{
+			StartTime: metav1.NewMicroTime(pod.StartedAt),
+			Ready:     mt.Manifest.PodReadinessMode() == model.PodReadinessIgnore || pod.AllContainersReady(),
+		}
+	case v1.PodSucceeded:
+		target.State.Terminated = &tiltruns.TargetStateTerminated{
+			StartTime: metav1.NewMicroTime(pod.StartedAt),
+		}
+	case v1.PodFailed:
+		podErr := strings.Join(pod.StatusMessages, "; ")
+		if podErr == "" {
+			podErr = fmt.Sprintf("Pod %q failed", pod.PodID.String())
+		}
+		target.State.Terminated = &tiltruns.TargetStateTerminated{
+			StartTime: metav1.NewMicroTime(pod.StartedAt),
+			Error:     podErr,
+		}
+	}
+
+	if target.State.Terminated == nil || target.State.Terminated.Error == "" {
+		for _, ctr := range pod.AllContainers() {
+			if ctr.Status == model.RuntimeStatusError {
+				target.State.Terminated = &tiltruns.TargetStateTerminated{
+					StartTime: metav1.NewMicroTime(pod.StartedAt),
+					Error:     fmt.Sprintf("Pod %s in error state: %s", pod.PodID, pod.Status),
+				}
+			}
+		}
+	}
+
+	// default to pending
```

Comment:
[maybe breaking] eek I believe this will leave with a pending k8sRuntimeTarget even if the resource itself is auto_init=False, and thus cause CI to hang forever, right?

Reply:
to put it another way: does `TestExitControlCI_DontBlockOnAutoInitFalse` fail if it has a _k8s resource_ with `auto_init=false`? (currently it's local resources with `auto_init=false`.)

Reply:
I bet the answer is yes, it's going to cause a hang 😢 

Good catch - I'll add a test case and fix

Reply:
Yuuuppp - fixed in 46684586f and split this test case into a K8s and local one, each of which goes through the full gamut of `TriggerMode`s now 🎉